flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/functions/AsyncTableFunction.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.functions;
+
+import org.apache.flink.annotation.Experimental;
+import org.apache.flink.api.common.functions.InvalidTypesException;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.typeutils.TypeExtractor;
+import org.apache.flink.streaming.api.functions.async.ResultFuture;
```

Comment:
If we add this new user function, we should make sure that the interface only contains things that are available in `flink-core` and/or `flink-table-common`. This is an interface that is specific to `flink-streaming`

Reply:
Yes. I think `ResultFuture` is a simple interface and can be moved to `flink-core`. The only concern is that is this a strong reason to move it? Another way is introducing a table's `ResultFuture`. But I prefer the first approach. What do you think?